cpp/tests/traversal/ms_bfs_test.cpp

```diff
@@ -0,0 +1,296 @@
+      for (size_t i = 0; i < h_sources.size(); i++) {
+        d_distances_ref[i] = cugraph::test::replace<vertex_t>(handle,
+                                                              std::move(d_distances_ref[i]),
+                                                              std::numeric_limits<vertex_t>::max(),
+                                                              static_cast<vertex_t>(0));
+
+        ref_sum +=
+          cugraph::test::reduce(handle, std::move(d_distances_ref[i]), static_cast<vertex_t>(0));
+      }
+
+      d_distances = cugraph::test::replace<vertex_t>(handle,
+                                                     std::move(d_distances),
+                                                     std::numeric_limits<vertex_t>::max(),
+                                                     static_cast<vertex_t>(0));
+
+      vertex_t ms_sum =
+        cugraph::test::reduce(handle, std::move(d_distances), static_cast<vertex_t>(0));
+
+      auto d_vertex_degree = graph_view.compute_out_degrees(handle);
+
+      d_sources = cugraph::test::sort<vertex_t>(handle, std::move(d_sources));
+
+      size_t seeds_degree_size = 0;
+
+      std::tie(d_vertex_degree, d_vertices, seeds_degree_size) =
+        cugraph::test::partition<vertex_t, vertex_t>(
+          handle, std::move(d_vertex_degree), std::move(d_vertices), d_sources);
+
+      d_vertex_degree.resize(seeds_degree_size, handle.get_stream());
+
+      auto seeds_degree_sum =
+        cugraph::test::reduce(handle, std::move(d_vertex_degree), static_cast<vertex_t>(0));
+
+      // Check the degree of the seed vertices
+      // If degree of all seeds is zero hence ref_sum is zero otherwise ref_sum > 0
+      ASSERT_TRUE(seeds_degree_sum ? ref_sum > 0 : ref_sum == 0);
+      ASSERT_TRUE(ref_sum < std::numeric_limits<vertex_t>::max());
+      ASSERT_TRUE(ref_sum == ms_sum);
+    }
+  }
+};
+
+using Tests_MsBfs_File = Tests_MsBfs<cugraph::test::File_Usecase>;
+using Tests_MsBfs_Rmat = Tests_MsBfs<cugraph::test::Rmat_Usecase>;
+
+TEST_P(Tests_MsBfs_File, CheckInt32Int32Float)
+{
+  run_current_test<int32_t, int32_t, float>(
+    override_File_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_MsBfs_File, CheckInt64Int64Float)
+{
+  run_current_test<int64_t, int64_t, float>(
+    override_File_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_MsBfs_Rmat, CheckInt32Int32Float)
+{
+  run_current_test<int32_t, int32_t, float>(
+    override_Rmat_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_MsBfs_Rmat, CheckInt64Int64Float)
+{
+  run_current_test<int64_t, int64_t, float>(
+    override_Rmat_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+INSTANTIATE_TEST_SUITE_P(
+  simple_test,
+  Tests_MsBfs_File,
+  ::testing::Combine(
+    // enable correctness checks
+    ::testing::Values(MsBfs_Usecase{2, 5, false, true}, MsBfs_Usecase{4, 9, true, true}),
+    ::testing::Values(cugraph::test::File_Usecase("test/datasets/netscience.mtx"),
```

Comment:
Does netscience have multiple components?

Reply:
Yes it has 396 components